### PR TITLE
[Feature] Support FA1.2

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description
This PR adds support for FA1.2 transferring and approving, and fixes as well FA2 responsive which was broken

### CODE REVIEW

File: context/fa1_2.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  targetAddress,
+  amount,
+  fa1_2Address,
+}: transfer) => `{ 
+    DROP ;
+    PUSH address "${fa1_2Address}" ;
+    CONTRACT %transfer (pair (address :from) (pair (address :to) (nat :amount))) ;
+    IF_NONE { PUSH string "contract dosen't exist" ; FAILWITH } { } ;
+    PUSH mutez 0 ;
+    PUSH  (pair (address :from) (pair (address :to) (nat :amount))) (pair "${walletAddress}" (pair "${targetAddress}" ${amount})) ;
```
Comment (rueshyna) [inline context/fa1_2.ts:40]:
```suggestion
    PUSH  (pair (address :from) (pair (address :to) (nat :amount))) (Pair "${walletAddress}" (Pair "${targetAddress}" ${amount})) ;
```

**Comment by rueshyna** [comment]
I misunderstood something. About decimals, FA2 token also need to apply decimals. You can get USDT from marigold faucet for testing. (I thought it was FA1.2)

File: components/ErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const ErrorMessage = ({ name }: props) => {
+  const [_, { error, touched }, __] = useField(name);
+
+  console.log(error);
+
+  return (
+    <span className="mt-1 inline-block break-words text-transparent">
+      {(!error || !touched) && "i"}
+
+      {touched && renderError(error, false, true)}
```
[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply components/ErrorMessage.tsx:15]: Hum, I don't think it's a requirement, but I guess that we can. `ErrorMessage` should be used in a form context, and `renderError` in the other cases

File: components/formUtils.tsx [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
+  withMargin: boolean = false,
+  asSpan: boolean = false
+) =>
+  asSpan ? (
+    <span
+      className={`${withMargin ? "mt-1" : ""} italic ${
+        !message ? "text-transparent" : "text-red-600"
+      }`}
+    >
+      {!message ? "a" : message}
```
[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply components/formUtils.tsx:12]: The point of this `a` is to keep having a letter, so the error message still have an height, and it avoids that the UI moves when an error message appears, also the `a` isn't visible as there's a `text-transparent`

[the remark this answers is not in the document]
Comment (rueshyna) [reply components/formUtils.tsx:12]: I see. Then,  can we call it `phantom`? (kind of latex style :stuck_out_tongue:)

File: components/FA2Transfer.tsx [hunk 4]
```diff
@@ -325,27 +314,18 @@ type formValue = {
...
+const FA2TransferGroup = ({ proposalIndex, remove }: props) => {
+  const { setFieldValue, getFieldProps } = useFormikContext();
+
   const [selectedTokens, setSelectedTokens] = useState<
     (fa2Token | undefined)[]
   >([]);
   const [additionalTransfers, setAdditionalTransfers] = useState<string[]>([]);
   const [contractAddress, setContractAddress] = useState("");
 
+  console.log(selectedTokens);
```
Comment (rueshyna) [inline components/FA2Transfer.tsx:328]: need to remove.

File: components/FA1_2.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    setFieldValue(makeName("tokenId"), newToken?.token.tokenId ?? "");
+    setFieldValue(
+      makeName("fa1_2Address"),
+      newToken?.token.contract.address ?? ""
+    );
+  };
+
+  const fetchTokens = useCallback(
+    (value: string, offset: number) =>
+      fetch(
```
Comment (rueshyna) [inline components/FA1_2.tsx:105]: How does the following suggestion and effort? We can do in next PR (I'd like to we start parsing lambda task ASAP.)

can you make retry (maybe 5 times) here, it the call doesn't get response in maybe 3 seconds? If no response, then show something like "server is no response, please retry later."

**Review by rueshyna — APPROVED** [review]

**Comment by rueshyna** [comment]
approved PR, but need to rebased to fix conflicts.

**Pull request merged**